--- NLO/SubProcesses/fks_singular.f.orig	2021-01-14 09:08:28.216954859 +0100
+++ NLO/SubProcesses/fks_singular.f	2021-01-14 09:06:13.450217667 +0100
@@ -112,7 +112,7 @@
       double precision wgtborn, alphas
       ! switch on/off here
       logical include_6to5_cnt 
-      data include_6to5_cnt /.false./ 
+      data include_6to5_cnt /.true./ 
 
 CMZMZ REMEMBER!!!!
 c     wgt1 : weight of the contribution not multiplying a scale log
@@ -126,7 +126,7 @@
 
       ! skip if we don't want this piece or if the scale is
       ! below mt
-      if (.not.include_6to5_cnt.or.scale.lt.mdl_mt) return
+      if (.not.include_6to5_cnt.or.scale.lt.mdl_mb) return
 
 C the contribution is the following (if mu > mt):
 C      Add a term -alphas n TF/3pi log (muR^2/mt^2) sigma(0) 
@@ -162,7 +162,7 @@
      &    - alphas / 3d0 / pi * TF * dble(alphasbpow) * amp_split(iamp) 
         
           amp_split_6to5f(orders_to_amp_split_pos(orders)) = 
-     &    dlog(qes2/mdl_mt**2) * 
+     &    dlog(qes2/mdl_mb**2) * 
      &     (alphas / 3d0 / pi * TF * dble(niglu)   
      &    - alphas / 3d0 / pi * TF * dble(alphasbpow)) * amp_split(iamp)
         endif
