--- a/SubProcesses/setscales.f
+++ b/SubProcesses/setscales.f
@@ -581,6 +581,11 @@
 cc      in the run_card (run_card.dat)                                           cc
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
-         tmp = user_dynamical_scale(PP)
+         tmp=0d0
+         do i=3,4
+           tmp=tmp+dsqrt(max(0d0,(pp(0,i)+pp(3,i))*(pp(0,i)-pp(3,i))))
+         enddo
+         tmp=tmp/4d0
+         temp_scale_id='H_T/4 := sum_i mT(i)/4, i=final state'
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
 cc      USER-DEFINED SCALE: END OF USER CODE                                     cc
 ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
